bundles/org.platformio.eclipse.ide.home.ui/src/org/platformio/eclipse/ide/home/internal/ui/terminal/TerminalOutput.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2021 PlatformIO and ArSysOp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Contributors:
+ *     Nikifor Fedorov (ArSysOp) - initial API and implementation
+ *******************************************************************************/
+package org.platformio.eclipse.ide.home.internal.ui.terminal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import org.eclipse.ui.console.IOConsoleOutputStream;
+import org.platformio.eclipse.ide.home.api.Output;
+
+public final class TerminalOutput implements Output {
+
+	private final IOConsoleOutputStream output;
+
+	public TerminalOutput(IOConsoleOutputStream output) {
+		this.output = output;
+	}
+
+	@Override
+	public void output(InputStream stream) {
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(stream))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				output.write((line + "\n").getBytes()); //$NON-NLS-1$
+			}
+		} catch (final IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Override
+	public void error(InputStream stream) {
+		try (BufferedReader br = new BufferedReader(new InputStreamReader(stream))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				output.write((line + "\n").getBytes()); //$NON-NLS-1$
+			}
+		} catch (final IOException e) {
+			e.printStackTrace();
```

Comment:
I would rather throw than catch, otherwise you will never get notified about failures